ceremonies/src/lib.rs

```diff
@@ -1659,6 +1603,66 @@ impl<T: Config> Pallet<T> {
 			.unwrap_or_else(|_| Self::ceremony_reward())
 	}
 
+	fn issue_rewards(
+		cid: CommunityIdentifier,
+		cindex: CeremonyIndexType,
+		meetup_idx: MeetupIndexType,
+		meetup_participants: Vec<T::AccountId>,
+		participants_indices: Vec<usize>,
+	) {
+		let reward = Self::nominal_income(&cid);
+		for i in &participants_indices {
+			let participant = &meetup_participants[*i];
+			trace!(target: LOG, "participant merits reward: {:?}", participant);
+			if <encointer_balances::Pallet<T>>::issue(cid, participant, reward).is_ok() {
+				<ParticipantReputation<T>>::insert(
+					(&cid, cindex),
+					participant,
+					Reputation::VerifiedUnlinked,
+				);
+			}
+			sp_io::offchain_index::set(&reputation_cache_dirty_key(participant), &true.encode());
+		}
+
+		<IssuedRewards<T>>::insert((cid, cindex), meetup_idx, ());
+		info!(target: LOG, "issuing rewards completed");
+
+		Self::deposit_event(Event::RewardsIssued(
+			cid,
+			meetup_idx,
+			participants_indices.len() as u8,
+		));
+	}
+
+	fn gather_meetup_validation_data(
+		cid: CommunityIdentifier,
+		cindex: CeremonyIndexType,
+		meetup_participants: Vec<T::AccountId>,
+	) -> (Vec<u32>, Vec<Vec<usize>>) {
+		let mut participant_votes: Vec<u32> = vec![];
+		let mut participant_attestations: Vec<Vec<usize>> = vec![];
+
+		// gather votes and attestations
+		for participant in meetup_participants.iter() {
+			let attestations = match Self::attestation_registry(
+				(&cid, cindex),
+				&Self::attestation_index((cid, cindex), &participant),
+			) {
+				Some(attestees) => attestees,
+				None => vec![],
+			};
+			// convert AccountId to local index
+			let attestation_indices = attestations
+				.into_iter()
+				.filter_map(|p| meetup_participants.iter().position(|s| *s == p))
+				.collect();
+
+			participant_attestations.push(attestation_indices);
```

Comment:
We can skip the collect here:

```
		let attestation_indices = attestations
				.into_iter()
				.filter_map(|p| meetup_participants.iter().position(|s| *s == p));

		participant_attestations.extend(attestation_indices);
```

Reply:
`participant_attestations: Vec<Vec<usize>>`
so i want to push a vec to a vec and not extend it

Reply:
Ok, I see. By bad. :)